src/js/ripple/amount.js

```diff
@@ -659,96 +462,72 @@ Amount.prototype.invert = function() {
  *
  * The regular expression below matches above cases, broken down for better understanding:
  *
- * ^\s*                         // start with any amount of whitespace
  * ([A-z]{3}|[0-9]{3})          // either 3 letter alphabetic currency-code or 3 digit numeric currency-code. See ISO 4217
- * \s*                          // any amount of whitespace
- * (-)?                         // optional dash
- * (\d+)                        // 1 or more digits
- * (?:\.(\d*))?                 // optional . character with any amount of digits
- * \s*                          // any amount of whitespace
- * ([A-z]{3}|[0-9]{3})?         // either 3 letter alphabetic currency-code or 3 digit numeric currency-code. See ISO 4217
- * \s*                          // any amount of whitespace
  * $                            // end of string
  *
  */
-Amount.human_RE_hex = /^\s*(-)?(\d+)(?:\.(\d*))?\s*([a-fA-F0-9]{40})\s*$/;
-Amount.human_RE = /^\s*([A-z]{3}|[0-9]{3})?\s*(-)?(\d+)(?:\.(\d*))?\s*([A-z]{3}|[0-9]{3})?\s*$/;
 
 Amount.prototype.parse_human = function(j, opts) {
   opts = opts || {};
 
-  var integer;
-  var fraction;
-  var currency;
-  var precision  = null;
-
-  // first check if it's a hex formatted currency
-  var matches = String(j).match(Amount.human_RE_hex);
-  if (matches && matches.length === 5 && matches[4]) {
-    integer  = matches[2];
-    fraction = matches[3] || '';
-    currency = matches[4];
-    this._is_negative = Boolean(matches[1]);
-  }
+  var hex_RE = /^[a-fA-F0-9]{40}$/;
+  var currency_RE = /^([a-zA-Z]{3}|[0-9]{3})$/;
+  function isNum(x) { return !isNaN(x); }
```

Comment:
I would prefer `isNumber`. Also, `Infinite` is a valid number, which we don't want here. `isNumber` in serializedtypes does a similar thing.